src/test/java/cat/udl/eps/entsoftarch/webingogeiapi/steps/EditGameStepDefs.java

```diff
@@ -0,0 +1,212 @@
+package cat.udl.eps.entsoftarch.webingogeiapi.steps;
+
+import cat.udl.eps.entsoftarch.webingogeiapi.domain.Game;
+import cat.udl.eps.entsoftarch.webingogeiapi.repository.GameRepository;
+import com.jayway.jsonpath.JsonPath;
+import cucumber.api.PendingException;
+import cucumber.api.java.en.And;
+import cucumber.api.java.en.Given;
+import cucumber.api.java.en.Then;
+import cucumber.api.java.en.When;
+import org.json.JSONException;
+import org.json.JSONObject;
+import org.junit.Assert;
+import org.skyscreamer.jsonassert.comparator.JSONComparator;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.MediaType;
+
+import java.time.ZonedDateTime;
+
+import static org.hamcrest.Matchers.endsWith;
+import static org.hamcrest.Matchers.is;
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.*;
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.get;
+import static org.springframework.test.web.servlet.result.MockMvcResultHandlers.print;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.jsonPath;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.status;
+import static sun.audio.AudioPlayer.player;
+
+
+public class EditGameStepDefs {
+
+    @Autowired
+    private final StepDefs stepDefs;
+    GameRepository gameRepository;
+    String idc;
+
+    public EditGameStepDefs(StepDefs stepDefs) {
+        this.stepDefs = stepDefs;
+    }
+
+    @Then("^The game with id (\\d+) has not been edited$")
+    public void theGameWithIdHasNotBeenEdited(int id) throws Exception {
+        stepDefs.result = stepDefs.mockMvc.perform(
+                get("/games/{id}", id)
+                        .accept(MediaType.APPLICATION_JSON)
+                        .with(AuthenticationStepDefs.authenticate()))
+                .andDo(print());
+    }
+
+
+    @Given("^There is a game with the pricePerCard (\\d+.\\d+) and id (\\d+)$")
+    public void thereIsAGameWithThePricePerCardAndId(double pricePerCard, int id) throws Exception {
+        JSONObject game = new JSONObject();
+        game.put("pricePerCard",pricePerCard);
+        stepDefs.result = stepDefs.mockMvc.perform(
+                post("/games/{id}",id)
+                        .contentType(MediaType.APPLICATION_JSON)
+                        .content(game.toString())
+                        .accept(MediaType.APPLICATION_JSON)
+                        .with(AuthenticationStepDefs.authenticate()))
+                .andDo(print());
+    }
+
+
+
+    @When("^I edit the game with id (\\d+)$")
+    public void iEditTheGameWithId(int id) throws Exception {
+        JSONObject game = new JSONObject();
+        game.put("id", id);
+        stepDefs.result = stepDefs.mockMvc.perform(
+                put("/games/{id}",id)
```

Comment:
There are no changes, in fact, by using PUT, you are erasing everything there was at that id by trying to update it with a Game that just has an id

Reply:
Consider PATCH and something to be changed